### ISSUE TITLE
ui5 build --all silently failing to build theme

### ISSUE THREAD

**Issue by srenwickblum** [SAP/ui5-builder#150]
## Expected Behavior
Our main "shared resources" project comprises of all the OpenUi5 modules, our own in-house modules and our own in-house theme, which is based on sap_belize.
Within the Theme project, `ui5 build` works perfectly. 
Within the main project, where all the dependancies are listed out (all openui5 modules, our own library and theme) it appears we need to call "ui5 build --all", which gathers the dependencies in node_modules and builds them.
This makes sense, since the npm versions of all OpenUi5 projects are not built. We plan to follow in these foot steps. 
We're hoping this (otherwise empty project) with all the dependancies, can be completely built with the command
`ui5 build --all `

## Current Behavior
However when calling "ui5 build --all", because our custom theme is based on belize (and therefore has a dependancy to belize), the belize theme is being built.

1. This is more than we want (need) - but we could live it with. 
2. In the attached example, although belize (and belize_plus) is being built, the custom theme is totally ignored (even though it builds perfectly when build is called with the theme project). 
3. In a sightly more complex variant where we have all the OpenUi5 modules included, I have observed only the sap_hcb is built, sap_belize (and friends) are also ignored. In our main build, we don't even get that far, belize gets built but belize_plus suddenly can't be built with errors like shown below. I hoped it was just one bug (oversight) and corrected it, but it just fails on something else, and goes on and on. Something isn't right but I can't track down what.
If there were tips, documentation on how to debug this stuff, I'd (maybe) be a little more helpful. 
````
info builder:builder 🛠  (13/15) Building project themelib_sap_belize
info builder:builder library themelib_sap_belize 🔨 (1/6) Running task replaceCopyright...
info builder:builder library themelib_sap_belize 🔨 (2/6) Running task replaceVersion...
info builder:builder library themelib_sap_belize 🔨 (3/6) Running task generateLibraryManifest...
info builder:builder library themelib_sap_belize 🔨 (4/6) Running task buildThemes...
ERR! builder:builder Build failed in 1.3 min
{ [Error: variable @sapUiText is undefined]
  type: 'Name',
  message: 'variable @sapUiText is undefined',
  filename: '/resources/sap/ui/core/themes/sap_belize_plus/shared.less',
  index: 135,
  line: 6,
  callLine: NaN,
  callExtract: undefined,
  stack: undefined,
  column: 8,
  extract: [ '#CSS_SCOPE_ROOT {', '\tcolor: @sapUiText;', '}' ] }

````

## Steps to reproduce the issue
1. Unzip the attached [demoProjects.zip](https://github.com/SAP/ui5-builder/files/2764239/demoProjects.zip)

2. call "npm link" in the theme project
3. call "npm link themelib_blum_mytheme" in the main_blum project
4. call npm update on both projects
5. from main_blum project call "ui5 build --all"

## Context
- UI5 Module Version *(output of `ui5 --version` when using the CLI)*: `{1.0}`
- Node.js Version: `{v8.9.4}`
- npm Version: `{5.6.0}`
- OS/Platform: {Windows}
- Browser *(if relevant)*: {n/a}
- Other information: {...}

## Affected components *(if known)*
- [x] [ui5-builder](https://github.com/SAP/ui5-builder)
- [ ] [ui5-server](https://github.com/SAP/ui5-server)
- [ ] [ui5-cli](https://github.com/SAP/ui5-cli)
- [x] [ui5-fs](https://github.com/SAP/ui5-fs)
- [ ] [ui5-project](https://github.com/SAP/ui5-project)
- [ ] [ui5-logger](https://github.com/SAP/ui5-logger)

## Log Output / Stack Trace
Attached [build_report.txt](https://github.com/SAP/ui5-builder/files/2764164/build_report.txt) is the redirected out put from doing a verbose build (where all dependencies are included).

**Comment by srenwickblum** [comment]
@RandomByte is there any update on this issue? Should I be looking at finding a work around for this?

**Comment by RandomByte** [comment]
Hey, sorry for the wait.

I'm having trouble reproducing this issue. First I couldn't built your theme from within the main_blum package. I solved this by changing the name fields of `main_blum/package.json` and `main_blum/ui5.yaml` from `themelib_blum_mytheme` to `main_blum`. I guess this naming wasn't intentional as it caused the tooling to ignore the actual theme project due to the duplicative naming.

Now the theme builds. Apparently without an error:

```
info builder:builder Building project main_blum including dependencies...
info builder:builder 🛠  (1/4) Building project sap.ui.core
info builder:builder library sap.ui.core 🔨 (1/7) Running task replaceCopyright...
info builder:builder library sap.ui.core 🔨 (2/7) Running task replaceVersion...
info builder:builder library sap.ui.core 🔨 (3/7) Running task generateLibraryManifest...
info builder:builder library sap.ui.core 🔨 (4/7) Running task generateLibraryPreload...
info builder:builder library sap.ui.core 🔨 (5/7) Running task buildThemes...
info builder:builder library sap.ui.core 🔨 (6/7) Running task createDebugFiles...
info builder:builder library sap.ui.core 🔨 (7/7) Running task uglify...
info builder:builder 🛠  (2/4) Building project themelib_sap_belize
info builder:builder library themelib_sap_belize 🔨 (1/7) Running task replaceCopyright...
info builder:builder library themelib_sap_belize 🔨 (2/7) Running task replaceVersion...
info builder:builder library themelib_sap_belize 🔨 (3/7) Running task generateLibraryManifest...
info builder:builder library themelib_sap_belize 🔨 (4/7) Running task generateLibraryPreload...
info builder:builder library themelib_sap_belize 🔨 (5/7) Running task buildThemes...
info builder:builder library themelib_sap_belize 🔨 (6/7) Running task createDebugFiles...
info builder:builder library themelib_sap_belize 🔨 (7/7) Running task uglify...
info builder:builder 🛠  (3/4) Building project themelib_blum_mytheme
info builder:builder library themelib_blum_mytheme 🔨 (1/7) Running task replaceCopyright...
info builder:builder library themelib_blum_mytheme 🔨 (2/7) Running task replaceVersion...
info builder:builder library themelib_blum_mytheme 🔨 (3/7) Running task generateLibraryManifest...
info builder:builder library themelib_blum_mytheme 🔨 (4/7) Running task generateLibraryPreload...
info builder:builder library themelib_blum_mytheme 🔨 (5/7) Running task buildThemes...
info builder:builder library themelib_blum_mytheme 🔨 (6/7) Running task createDebugFiles...
info builder:builder library themelib_blum_mytheme 🔨 (7/7) Running task uglify...
info builder:builder 🛠  (4/4) Building project main_blum
info builder:builder library main_blum 🔨 (1/7) Running task replaceCopyright...
info builder:builder library main_blum 🔨 (2/7) Running task replaceVersion...
info builder:builder library main_blum 🔨 (3/7) Running task generateLibraryManifest...
info builder:builder library main_blum 🔨 (4/7) Running task generateLibraryPreload...
info builder:builder library main_blum 🔨 (5/7) Running task buildThemes...
info builder:builder library main_blum 🔨 (6/7) Running task createDebugFiles...
info builder:builder library main_blum 🔨 (7/7) Running task uglify...
info builder:builder Build succeeded in 24 s
```

![screenshot 2019-02-04 at 17 54 40](https://user-images.githubusercontent.com/1589939/52223654-6377f980-28a6-11e9-8b6c-e20ac3b86b9b.png)


I'm using the latest UI5 CLI `v1.0.1`.

Your attached build_report.txt seems to be from a different setup as the dependency tree is way bigger than in your demo project.

Could you please re-check on how to reproduce this issue? Thanks!

**Comment by srenwickblum** [comment]
@RandomByte Thanks for looking into this. Your tip with v1.0.1 was the secret sauce, but not entirely. When I added the belieze dependancy to the package.json in the project where everything was built, then it worked correctly. My dependancy structure looks something like this

resource-project
    - blum-library
         - sap.ui.core (+friends)
    - blum-theme
         - sap.ui.core
         - belize_theme
         - blum-library

So I expected, when I added "blum-library" and "blum_theme" to the resource-project package.json, it would build everything. This wasn't the case, it actually failed on building the belize theme, failing to resolve variables. Adding all the sap libraries and the sap belize them to the resource-project cured this issue.

What remains is a very bizarre phantom issue where part of the output path is upper cased. But I think it makes sense to close this ticket and I'll try create a "simple as possible" example that exhibits this. 

Thanks again for your rapid (and very helpful) response!
